core/component.go

```diff
@@ -118,6 +118,18 @@ func NewComponent(ctx log.Interface, serviceName string, announcedAddress string
 		go http.ListenAndServe(fmt.Sprintf(":%d", healthPort), nil)
 	}
 
+	if monitors := viper.GetStringMapString("monitor-servers"); len(monitors) != 0 {
+		component.Monitors = map[string]pb_noc.MonitorClient{}
```

Comment:
I think the convention is to use `make(map[string]...` if you initialize it as empty.
If you check for `nil` it's fine, but for other developers it's really easy to forget this.